src/main/java/com/aws/greengrass/mqttbridge/MQTTBridge.java

```diff
@@ -179,6 +164,65 @@ public void shutdown() {
         }
     }
 
+    public class CertificateAuthorityChangeHandler {
+
+        private BatchedSubscriber subscriber;
+
+        private void onCAChange() {
+            findCATopic()
+                    .map(Coerce::toStringList)
+                    .filter(certs -> !Utils.isEmpty(certs))
+                    .ifPresent(this::updateCA);
+        }
+
+        private void updateCA(List<String> certs) {
+            logger.atDebug().kv("numCaCerts", certs.size()).log("CA update received");
+            try {
+                mqttClientKeyStore.updateCA(certs);
+            } catch (IOException | CertificateException | KeyStoreException e) {
+                serviceErrored(e);
+            }
+        }
+
+        /**
+         * Begin listening and responding to CDA CA changes.
+         *
+         * <p>This operation is idempotent.
+         */
+        public void start() {
+            if (subscriber == null) {
+                Topic caTopic = findCATopic().orElse(null);
+                if (caTopic == null) {
+                    return;
+                }
+                subscriber = new BatchedSubscriber(caTopic, (what, whatChanged) -> onCAChange());
+            }
+            subscriber.subscribe();
+        }
+
+        /**
+         * Stop listening to CDA CA changes.
+         */
+        public void stop() {
+            if (subscriber != null) {
+                subscriber.unsubscribe();
+            }
+        }
+
+        private Optional<Topic> findCATopic() {
+            try {
+                return Optional.of(kernel
+                        .locate(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME).getConfig()
+                        .lookup(RUNTIME_STORE_NAMESPACE_TOPIC,
+                                ClientDevicesAuthService.CERTIFICATES_KEY,
+                                ClientDevicesAuthService.AUTHORITIES_TOPIC));
+            } catch (ServiceLoadException e) {
+                logger.atWarn().cause(e).log("unable to locate service");
```

Comment:
the log message should make sense to a customer. What is the impact of this warning, and what should they do about it. 

For example:

"Unable to locate `CLIENT_DEVICES_AUTH_SERVICE_NAME`. MQTT Bridge may be unable to connect to the broker. Ensure that `CLIENT_DEVICES_AUTH_SERVICE_NAME` component is deployed"

Reply:
done 👍 makes sense, will keep in mind for next time